src/rgw/rgw_rados.cc

```diff
@@ -5270,19 +5270,23 @@ int RGWRados::Object::Delete::delete_obj(optional_yield y, const DoutPrefixProvi
   store->remove_rgw_head_obj(op);
 
   auto& ioctx = ref.pool.ioctx();
+  BucketShard *bs = nullptr;
+  index_op.get_bucket_shard(&bs, dpp);
+  auto& index_ref = bs->bucket_obj.get_ref();
+
   r = rgw_rados_operate(dpp, ioctx, ref.obj.oid, &op, null_yield);
 
   /* raced with another operation, object state is indeterminate */
   const bool need_invalidate = (r == -ECANCELED);
 
-  int64_t poolid = ioctx.get_id();
+  int64_t poolid = index_ref.pool.ioctx().get_id();
```

Comment:
@Huber-ming can you please help me understand why this change is needed?

it looks like cls_rgw is expecting to get these values from the data pool where the head object resides, not from the index pool. `RGWRados::Object::Write::_do_write_meta()` gets these from the head object's pool as well: https://github.com/ceph/ceph/blob/master/src/rgw/rgw_rados.cc#L3239-L3240